pyanaconda/payload/manager.py

```diff
@@ -16,152 +15,131 @@
-    immediately trigger that listener.
-
     The payload thread data should be accessed using the payloadMgr object,
     and the running thread can be accessed using threadMgr with the
     THREAD_PAYLOAD constant, if you need to wait for it or something. The
     thread should be started using payloadMgr.restart_thread.
     """
 
-    def __init__(self):
-        self._event_lock = threading.Lock()
-        self._event_listeners = {}
-        self._thread_state = PayloadState.STARTED
-
-        # Initialize a list for each event state
-        for _name, value in PayloadState.__members__.items():  # pylint: disable=no-member
-            self._event_listeners[PayloadState(value)] = []
-
-    def add_listener(self, event_id, func):
-        """Add a listener for an event.
-
-        :param int event_id: The event to listen for, one of the EVENT_* constants
-        :param function func: An object to call when the event is reached
-        """
-
-        # Check that the event_id is valid
-        assert isinstance(event_id, PayloadState)
+    @property
+    def steps(self):
+        """Total number of steps."""
+        return 1
 
-        # Add the listener inside the lock in case we need to run immediately,
-        # to make sure the listener isn't triggered twice
-        with self._event_lock:
-            self._event_listeners[event_id].append(func)
+    @property
+    def is_running(self):
+        """Is the payload thread running right now?"""
+        return threadMgr.exists(THREAD_PAYLOAD_RESTART) or threadMgr.exists(THREAD_PAYLOAD)
 
-            # If an error event was requested, run it if currently in an error state
-            if event_id == PayloadState.ERROR:
-                if event_id == self._thread_state:
-                    func()
-            # Otherwise, run if the requested event has already occurred
-            elif event_id <= self._thread_state:
-                func()
+    def restart_thread(self, *args, **kwargs):
+        """Start or restart the payload thread."""
+        self.start(*args, **kwargs)
 
-    def restart_thread(self, payload, fallback=False, try_media=True, only_on_change=False):
+    def start(self, *args, **kwargs):
         """Start or restart the payload thread.
 
         This method starts a new thread to restart the payload thread, so
         this method's return is not blocked by waiting on the previous payload
         thread. If there is already a payload thread restart pending, this method
         has no effect.
 
-        :param payload.Payload payload: The payload instance
-        :param bool fallback: Whether to fall back to the default repo in case of error
-        :param bool try_media: Whether to check for valid mounted media
-        :param bool only_on_change: Restart thread only if existing repositories changed.
-            This won't restart thread even when a new repository was added!!
+        This won't restart thread even when a new repository was added!
```

Comment:
I wonder this comment when two lines below is ``log.debug("Restarting payload thread")``.

Reply:
It is a leftover. I will remove it.